File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.py
def solution(nums):
    results = []
    for i in range(len(nums)):
        for j in range(i + 1, len(nums)):
            for k in range(j + 1, len(nums)):
                results.append(nums[i] + nums[j] + nums[k])
    results.sort()
    sosu = []
    for i in results:
        count = 0
        for j in range(1, i + 1):
            if i % j == 0:
                count += 1
        if count == 2:
            sosu.append(i)
    answer = len(sosu)
    return answer

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.java
class Solution {
    public int solution(int[] nums) {
        int max = 0;
        for (int num : nums) {
            if (num > max) max = num;
        }
        int n = nums.length;
        boolean[] sosu = new boolean[3 * max];
        // 체 만들기 - false면 소수
        sosu[0] = true;
        sosu[1] = true;
        for (int i = 2; i < 3 * max; i++) {
            if (!sosu[i]) {
                int m = 2;
                while (i * m < 3 * max) {
                    sosu[i * m] = true;
                    m++;
                }
            }
        }
        int count = 0;
        for (int i = 0; i < n - 2; i++) {
            for (int j = i + 1; j < n - 1; j++) {
                for (int k = j + 1; k < n; k++) {
                    if (!sosu[nums[i] + nums[j] + nums[k]]) {
                        count++;
                    }
                }
            }
        }
        return count;
    }
}